src/js/spotify.js

```diff
@@ -0,0 +1,418 @@
+    GetSavedShowsUrl: function() { return `https://api.spotify.com/v1/me/shows`; },
+    GetCurrentUserPlaylistsUrl: function() { return `https://api.spotify.com/v1/me/playlists`; },
+    GetNewAlbumReleasesUrl: function() { return `https://api.spotify.com/v1/browse/new-releases`; },
+    GetFeaturedPlaylistsUrl: function() { return `https://api.spotify.com/v1/browse/featured-playlists`; },
+    GetCurrentUserProfileUrl: function() { return `https://api.spotify.com/v1/me`; }
+
+};
+
+const subscriptions = {
+    error: {
+        name: "",
+        desc: "Ошибка при определении подписки."
+    },
+    open: {
+        name: "Бесплатно",
+        desc: "Треки можно слушать вперемешку. Есть реклама."
+    },
+    premium: {
+        name: "Премиум",
+        desc: "Премиум подписка. Доступен весь функционал."
+    }
+}
+
+/**
+ * Returns name of subscription by provided sub_name from Spotify response.
+ * @param  {} sub_name - Subscription name.
+ */
+export function GetSubscriptionName(sub_name) {
+    if (subscriptions.hasOwnProperty(sub_name))
+        return subscriptions[sub_name].name;
+    else
+        return subscriptions["error"].name;
+}
+
+/**
+ * Returns description of subscription by provided sub_name from Spotify response.
+ * @param  {} sub_name - Subscription name.
+ */
+export function GetSubscriptionDesc(sub_name) {
+    if (subscriptions.hasOwnProperty(sub_name))
+        return subscriptions[sub_name].desc;
+    else
+        return subscriptions["error"].desc;
+}
+
+/**
+ * Returns best result for search
+ * @param  {} data - Returned response result in JSON format from search request.
+ * @returns Object with fields: href, name, image_url, type.
+ */
+function GetBestItemBySearch(data) {
+    if (data.artists.items.length === 0 && data.tracks.items.length === 0)
+        return null;   
+    if (data.artists.items.length >= 1) {
+        let image_link = data.artists.items[0].images[0].url;
+        return {
+            href: data.artists.items[0].href,
+            name: data.artists.items[0].name,
+            image_url: image_link === undefined ? defaultAvatar : image_link,
+            type: 'Исполнитель'
+        };
+    }
+    else  {
+        let image_link = data.tracks.items[0].album.images[0].url;
+        return {
+            href: data.tracks.items[0].href,
+            name: data.tracks.items[0].name,
+            image_url: image_link === undefined ? defaultAvatar : image_link,
+            type: 'Трек'
+        };
+    }
+}
+
+/**
+ * Return tracks for search
+ * @param  {} data - Returned response result in JSON format from search request.
+ * @returns Array with track objects with fields: href, name, image_url, type, author, duration
+. */
+function GetTracksFromSearch(data) {
+    let result = [];
+    for (let i = 0; i < data.tracks.items.length; i++) {
+        if (result.length === search_tracks_limit)
+            break;
+        let image_link = data.tracks.items[i].album.images.length > 0 ?
```

Comment:
переменная нигде не изменяется.
она должна быть const
```suggestion
        const image_link = data.tracks.items[i].album.images.length > 0 ?
```

Reply:
Избавился от этой переменной